src/AppInstallerCommonCore/NameNormalization.cpp

```diff
@@ -501,4 +501,28 @@ namespace AppInstaller::Utility
     {
         return m_normalizer->NormalizePublisher(publisher);
     }
+
+    std::string NormalizedName::GetNormalizedName(NormalizationField fieldsToInclude) const
+    {
+        std::string result = Name();
+
+        if (WI_IsFlagSet(fieldsToInclude, NormalizationField::Architecture) && m_arch != Utility::Architecture::Unknown)
+        {
+            result += '(' + std::string(Utility::ToString(m_arch)) + ')';

```

Comment:
What about for packages that just include their architecture and not the parens? Or where it isn't just the arch to string? Won't this still cause issues? I feel like where DisplayName is included in the manifest, the normalization shouldn't occur.
```
  13  AppsAndFeaturesEntries:
  14:   - DisplayName: Bluebeam Revu x64 21
```
```
  34: - DisplayName: calibre 64bit
  35    UpgradeCode: '{5DD881FF-756B-4097-9D82-8C0F11D521EA}'
```
```
  23    - ProductCode: SyncBackPro64_is1_is1
  24:     DisplayName: SyncBackPro x64
```
```
  87:   - DisplayName: 7-Zip 22.01 (x64 edition)
  88      Publisher: Igor Pavlov
```
```
  28    - Publisher: Adobe
  29:     DisplayName: Adobe Acrobat DC (64-bit)
```

Reply:
Our normalization code uses a regex which should take care of different formats of arch representations. x64, (x64), 64bit etc.

The reason we still need to do normalization on DisplayName comes from various considerations:
1. Including every declared DisplayName explodes the index (which is currently around 13 mb), it will be at least several times bigger. Considering by default winget updates source every 3 mins if new version applicable, it's a lot of data downloading.
2. If DisplayName contains version (or any other Package Version specific string) and we don't normalize it, our correlation will rely on the repo containing every version of the package (with every package version declaring DisplayName). For example, if DisplayName is Foo v1.0.0 and Foo v2.0.0 (2 versions exist in winget-pkgs repo), and if we don't normalize them in index, a local installed version with Foo v1.5.0 will not be able to get correlated. We cannot guarantee winget-pkgs will have every version of a package listed and client should not rely on this assumption.

According to our brief analysis, the arch is the main source of false correlation. And it makes sense to keep arch info specifically. We could expand to include locale, or even version if we decide to in the future. But each expansion will come with the cost of making index multiple times larger, and less correlation success if the installed version is not in the repo. Unless we redesign the index structure in future winget v2 (if winget v2 is a thing).

Reply:
I see; thanks for the detailed explanation!